moonraker/components/zeroconf.py

```diff
@@ -32,20 +38,26 @@ async def unregister_services(self, infos: List[AsyncServiceInfo]) -> None:
         await asyncio.gather(*background_tasks)
         await self.aiozc.async_close()
 
-class ZeroconfRegistrar():
+
+class ZeroconfRegistrar:
     def __init__(self, config: ConfigHelper) -> None:
         self.server = config.get_server()
         self.event_loop = self.server.get_event_loop()
 
         host_name, port = self.server.get_host_info()
-        addresses = [socket.gethostbyname(host_name)]
+        addresses = [socket.inet_aton(self.server.host)]
+        if self.server.host == "0.0.0.0":
```

Comment:
We don't want to directly access attributes of other classes.  If necessary we can add a getter to the server class or refactor get_host_info to return the bound address.